chemprop/cli/train.py

```diff
@@ -621,12 +621,13 @@ def main(args):
         results = trainer.test(model, test_loader)[0]
         logger.info(f"Test results: {results}")
 
-    p_model = args.output_dir / "model.pt"
+    p_model = args.output_dir / "model.ckpt"
     torch.save(model.state_dict(), p_model)
     logger.info(f"model state dict saved to '{p_model}'")
```

Comment:
I think during today's chemprop v2 dev meeting, we think this can be removed.

Reply:
There's a bit of an over-reliance on `load_from_checkpoint`, a bit like using a sledgehammer to crack a nut. It solves the problem for loading model state to make predictions, but it muddies the water between a "checkpoint" and "model state." The former contains all the information to continue training, and the latter contains just the weight. Of course, you can't just load weights without knowing the model's architecture, so we need to load that elegantly as well, at which point you might as well just couple them together when saving because one is fairly useless without the other.

To summarize, we should think about loading "model state + arch" when testing rather than assuming we have a checkpoint and just relying on `load_from_checkpoint` to do the trick. It might help first to copy code from `load_from_checkpoint` into a separate `@classmethod`, `load_model`, that loads both model architecture ("hyperparamters") and weights. It should be assumed that a "checkpoint" is a superset of a "model state" because, again, the former contains a model's state "and then some."

Reply:
Thanks for the comments, @KnathanM and @davidegraff. I think it's best to add the `load_model` functionality in a separate PR. I will open up an issue to keep track of this.